chasm/field.go

```diff
@@ -58,10 +60,37 @@ func (fi fieldInternal) IsEmpty() bool {
 	return fi.value == nil
 }
 
-func (f Field[T]) Get(Context) (T, error) {
-	// remember to handle d == nil case
+func (f Field[T]) Get(chasmContext Context) (T, error) {
+	// If value is not nil, use it.
+	if f.Internal.value != nil {
+		return f.Internal.value.(T), nil
+	}
 
-	panic("not implemented")
+	// If node is nil, then there is nothing to deserialize from, return nil.
+	if f.Internal.node == nil {
+		var v T
+		return v, nil
+	}
+
+	var err error
+	switch f.Internal.fieldType {
+	case fieldTypeComponent:
+		err = f.Internal.node.prepareComponentValue(chasmContext)
+	case fieldTypeData:
+		// For data fields, T is always a concrete type.
+		err = f.Internal.node.prepareDataValue(chasmContext, reflect.TypeFor[T]())
+	case fieldTypeComponentPointer:
+		panic("not implemented")
+	default:
+		err = serviceerror.NewInternal(fmt.Sprintf("unsupported field type: %v", f.Internal.fieldType))
+	}
+
+	if err != nil {
+		var nilT T
+		return nilT, err
+	}
+	f.Internal.value = f.Internal.node.value
```

Comment:
hmm I feel it's dangerous/error prone to keep two copies of the value. Is the assumption here that the value always have pointer type and two values fields will never go out of sync?

I was originally thinking the Internal.value & field type is only used for newly created field which doesn't have a backing node yet.

Reply:
Ok, I wasn't sure about it either. I introduced "getter" methods which look in `node` first and then into the `value` itself. It also simplified few other places.